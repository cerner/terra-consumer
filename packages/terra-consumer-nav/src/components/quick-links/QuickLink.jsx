import React from 'react';
import PropTypes from 'prop-types';
import Button from 'terra-button';
import classNames from 'classnames/bind';
import styles from './QuickLink.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The path the quicklink should lead to.
   */
  uri: PropTypes.string.isRequired,
  /**
   * The text displayed on the link.
   */
  text: PropTypes.string.isRequired,
};

const QuickLink = ({
  uri,
  text,
  ...customProps
}) => (
  <Button {...customProps} className={cx('quicklink', customProps.className)} href={uri} text={text} key={text} />
);

QuickLink.propTypes = propTypes;

export default QuickLink;
