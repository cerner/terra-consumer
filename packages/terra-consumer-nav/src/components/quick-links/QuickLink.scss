@import '../../variables';

:local {
  .quicklinks-container {
    margin: 0 25px;
  }

  .quicklink {
    background: var(--terra-consumer--quicklink-background, rgba(255, 255, 255, 0.9));
    border: 0;
    border-radius: 3px;
    color: var(--terra-consumer--quicklink-color, #3d3d3d);
    display: inline-block;
    font-family: inherit;
    font-size: 16px;
    line-height: 35px;
    margin-bottom: 10px;
    padding: 0;
    text-align: center;
    text-decoration: none;
    width: 100%;

    &:hover {
      background: var(--terra-consumer--quicklink-background-hover, #eee);
      cursor: pointer;
    }

    @media screen and (max-width: $terra-medium-breakpoint) {
      display: none;
    }
  }

  .quicklink-icon {
    padding-right: 5px;
  }
}
