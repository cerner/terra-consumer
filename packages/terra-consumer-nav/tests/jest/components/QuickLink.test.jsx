import React from 'react';
import QuickLink from '../../../src/components/quick-links/QuickLink';

const testData = {
  uri: '#testPath',
  text: 'testIcon',
};

describe('QuickLink', () => {
  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(<QuickLink {...testData} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should apply custom classes', () => {
    const wrapper = shallow(<QuickLink {...testData} className="test-class" />);
    expect(wrapper).toMatchSnapshot();
  });
});
