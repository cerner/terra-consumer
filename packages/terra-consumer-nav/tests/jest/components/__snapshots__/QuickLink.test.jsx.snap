// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickLink should apply custom classes 1`] = `
<Button
  className="quicklink test-class"
  href="#testPath"
  isBlock={false}
  isCompact={false}
  isDisabled={false}
  isReversed={false}
  text="testIcon"
  type="button"
  variant="default"
/>
`;

exports[`QuickLink should render a default component 1`] = `
<Button
  className="quicklink"
  href="#testPath"
  isBlock={false}
  isCompact={false}
  isDisabled={false}
  isReversed={false}
  text="testIcon"
  type="button"
  variant="default"
/>
`;
